prediction_market_agent/tools/mech/utils.py

```diff
@@ -0,0 +1,92 @@
+import os
+import tempfile
+import typing as t
+from contextlib import contextmanager
+from enum import Enum
+
+from mech_client.interact import ConfirmationType, interact
+from pydantic import BaseModel
+
+from prediction_market_agent.tools.mech.api_keys import MechAPIKeys
+from prediction_market_agent.tools.mech.mech.packages.polywrap.customs.prediction_with_research_report import (
+    prediction_with_research_report,
+)
+from prediction_market_agent.tools.mech.mech.packages.valory.customs.prediction_request import (
+    prediction_request,
+)
+from prediction_market_agent.utils import completion_str_to_json
+
+
+class MechResult(BaseModel):
+    p_yes: float
+    p_no: float
+    confidence: float
+    info_utility: float
+
+
+@contextmanager
+def saved_str_to_tmpfile(s: str) -> t.Iterator[str]:
+    # Write the string to the temporary file
+    with tempfile.NamedTemporaryFile(delete=False) as tmp:
+        tmp.write(s.encode())
+
+    yield tmp.name
+
+    # Finally remove the temporary file
+    os.remove(tmp.name)
+
+
+class MechTool(str, Enum):
+    PREDICTION_WITH_RESEARCH_REPORT = "prediction-with-research-conservative"
+    PREDICTION_ONLINE = "prediction-online"
+
+
+def mech_request(question: str, mech_tool: MechTool) -> MechResult:
+    private_key = MechAPIKeys().bet_from_private_key.get_secret_value()
+    with saved_str_to_tmpfile(private_key) as tmpfile_path:
+        # Increase gas price to reduce chance of 'out of gas' transaction failures
+        mech_strategy_env_var = "MECHX_LEDGER_DEFAULT_GAS_PRICE_STRATEGY"
+        if os.getenv(mech_strategy_env_var):
+            raise ValueError(f"{mech_strategy_env_var} already set in the environment.")
+        os.environ[mech_strategy_env_var] = "gas_station"
+
+        response = interact(
+            prompt=question,
+            # Taken from https://github.com/valory-xyz/mech?tab=readme-ov-file#examples-of-deployed-mechs
+            agent_id=6,
+            private_key_path=tmpfile_path,
+            # To see a list of available tools, comment out the tool parameter
+            # and run the function. You will be prompted to select a tool.
+            tool=mech_tool.value,
+            confirmation_type=ConfirmationType.WAIT_FOR_BOTH,
+        )
+        del os.environ[mech_strategy_env_var]
```

Comment:
Why deleting this? I assume mech writes this, right?

Reply:
This env var is read inside the `interact` function. So we set it before calling. We unset it after so as not to pollute the env vars